R/zchunk_L244.building_det.R

```diff
@@ -0,0 +1,1044 @@
+
+    L244.Floorspace_resid <- L144.flsp_bm2_R_res_Yh %>%
+      mutate(base.building.size = round(value, energy.DIGITS_FLOORSPACE)) %>%
+      select(-value) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      mutate(gcam.consumer = A44.gcam_consumer_resid$gcam.consumer,
+             nodeInput = A44.gcam_consumer_resid$nodeInput,
+             building.node.input = A44.gcam_consumer_resid$building.node.input) %>%
+      select(region, gcam.consumer, nodeInput, building.node.input, year, base.building.size)
+
+    # Commercial floorspace
+    A44.gcam_consumer_comm <- A44.gcam_consumer %>%
+      filter(grepl("comm", A44.gcam_consumer$gcam.consumer))
+
+    L244.Floorspace_comm <- L144.flsp_bm2_R_comm_Yh %>%
+      mutate(base.building.size = round(value, energy.DIGITS_FLOORSPACE)) %>%
+      select(-value) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      mutate(gcam.consumer = A44.gcam_consumer_comm$gcam.consumer,
+             nodeInput = A44.gcam_consumer_comm$nodeInput,
+             building.node.input = A44.gcam_consumer_comm$building.node.input) %>%
+      select(region, gcam.consumer, nodeInput, building.node.input, year, base.building.size)
+
+    L244.Floorspace <- bind_rows(L244.Floorspace_resid, L244.Floorspace_comm) %>%
+      filter(year %in% BASE_YEARS)
+
+    # Demand function
+    # L244.DemandFunction_serv and L244.DemandFunction_flsp: demand function types
+    L244.DemandFunction_serv <- write_to_all_regions(A44.demandFn_serv, LEVEL2_DATA_NAMES[["DemandFunction_serv"]],
+                                                     GCAM_region_names = GCAM_region_names)
+    L244.DemandFunction_flsp <- write_to_all_regions(A44.demandFn_flsp, LEVEL2_DATA_NAMES[["DemandFunction_flsp"]],
+                                                     GCAM_region_names = GCAM_region_names)
+
+    # Floorspace demand satiation
+    # L244.Satiation_flsp: Satiation levels assumed for floorspace
+    L244.Satiation_flsp_class <- A44.satiation_flsp %>%
+      gather(sector, value, resid, comm) %>%
+      mutate(satiation.level = value * CONV_THOUS_BIL) %>%
+      select(-value)
+
+    L244.Satiation_flsp <- write_to_all_regions(A44.gcam_consumer, c("region", "gcam.consumer", "nodeInput", "building.node.input"), # replace with LEVEL2_DATA_NAMES[["BldNodes]]
+                                                GCAM_region_names = GCAM_region_names) %>%
+      # Match in the region class, and use this to then match in the satiation floorspace
+      left_join_error_no_match(A_regions %>% select(region, region.class),
+                               by = "region") %>%
+      left_join_error_no_match(L244.Satiation_flsp_class, by = c("region.class", "gcam.consumer" = "sector")) %>%
+      select(LEVEL2_DATA_NAMES[["Satiation_flsp"]])
+
+    # Satiation adder - Required for shaping the future floorspace growth trajectories in each region
```

Comment:
Should this ("Satiation adder") be defined more clearly?

Reply:
Maybe, but I would need @pkyle to help with that

Reply:
The satiation adder allows the starting (final calibration year) position of any region and sector to be set along the satiation demand function. It was initially intended to be a non-zero, positive "subsistence" requirement (so that as GDP approaches 0, floorspace per capita doesn't do the same). In its current usage, it is calculated so that the mid-point of the floorspace satiation demand function to an exogenous level of per-capita GDP. And it is negative in a number of regions.
The unit is the same as per-capita floorspace - million square meters per person (unconverted from the model's units of billion square meters and thousand persons). So, it has a physical unit but not a physical meaning!